docs/projects/run-build-deploy.md

```diff
@@ -147,20 +161,90 @@ image that was set when the function was added to the project.
 
 For example:
 
-    project = mlrun.new_project(project_name, "./proj")
-    # use v1 of a pre-built image as default
-    project.set_default_image("myrepo/my-prebuilt-image:v1")
-    # set function without an image, will use the project's default image
-    project.set_function("mycode.py", "prep")
+```python
+ project = mlrun.new_project(project_name, "./proj")
+ # use v1 of a pre-built image as default
+ project.set_default_image("myrepo/my-prebuilt-image:v1")
+ # set function without an image, will use the project's default image
+ project.set_function("mycode.py", "prep")
+
+ # function will run with the "myrepo/my-prebuilt-image:v1" image
+ run1 = project.run_function("prep", params={"x": 7}, inputs={'data': data_url})
+
+ ...
+
+ # replace the default image with a newer v2
+ project.set_default_image("myrepo/my-prebuilt-image:v2")
+ # function will now run using the v2 version of the image 
+ run2 = project.run_function("prep", params={"x": 7}, inputs={'data': data_url})
+```
+
+<a id="build_config"></a>
+## Image build configuration
+
+Using the {py:meth}`~mlrun.projects.MlrunProject.set_default_image` function configures a project to use an existing 
+image. In addition to that, a project can contain the configurations for building this default image. This is done 
+through the {py:meth}`~mlrun.projects.MlrunProject.build_config` and {py:meth}`~mlrun.projects.MlrunProject.build_image` 
+functions. 
+
+The project build configuration is maintained in the project object. When saving, exporting and importing the project 
+these configurations are carried over with it. This makes it simple to transport a project between systems while 
+ensuring the needed runtime images are built and are ready for execution. 
+
+When using {py:meth}`~mlrun.projects.MlrunProject.build_config` build configurations can be passed along with the 
+resulting image name, and these are used to build the image. The image name is assigned following these rules:
+
+1. The name passed in the `image` parameter of the {py:meth}`~mlrun.projects.MlrunProject.build_config` function.
+2. The project's default image name, if configured using {py:meth}`~mlrun.projects.MlrunProject.set_default_image`.
+3. The value set in MLRun's `default_project_image_name` config parameter - by default this value is 
+   `.mlrun-project-image-{name}` with the project name as template parameter.
+
+For example:
+
+```python
+ # Set image config for current project object, using base mlrun image with additional requirements. 
+ image_name = ".my-project-image"
+ project.build_config(
+     image=image_name,
+     set_as_default=True,
+     with_mlrun=False,
+     base_image="mlrun/mlrun",
+     requirements=["vaderSentiment"],
+ )
+
+ # Export project configuration. The yaml file will contain build configuration in it
```

Comment:
# Export the project configuration. The yaml file will contain the build configuration